src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -0,0 +1,902 @@
+        return this.currentLeg.currentWaypoint;
+    }
+
+    /**
+     * Return the next `LegModel`, if it exists
+     *
+     * @for RouteModel
+     * @property nextLeg
+     * @type {LegModel}
+     */
+    get nextLeg() {
+        if (!this.hasNextLeg()) {
+            return null;
+        }
+
+        return this._legCollection[1];
+    }
+
+    /**
+     * Return the next `WaypointModel`, from current or future leg
+     *
+     * @for RouteModel
+     * @property nextWaypoint
+     * @type {WaypointModel}
+     */
+    get nextWaypoint() {
+        if (!this.hasNextWaypoint()) {
+            return null;
+        }
+
+        if (this.currentLeg.hasNextWaypoint()) {
+            return this.currentLeg.nextWaypoint;
+        }
+
+        return this.nextLeg.currentWaypoint;
+    }
+
+    /**
+     * Return an array of all waypoints in all legs of the route
+     *
+     * @for RouteModel
+     * @property waypoints
+     * @type {array<WaypointModel>}
+     */
+    get waypoints() {
+        return _reduce(this._legCollection, (waypointList, legModel) => {
+            return waypointList.concat(legModel.waypoints);
+        }, []);
+    }
+
+    // ------------------------------ LIFECYCLE ------------------------------
+
+    /**
+     * Initialize class properties
+     *
+     * @for RouteModel
+     * @method init
+     * @param routeString {string}
+     * @chainable
+     */
+    init(routeString) {
+        this._legCollection = this._generateLegsFromRouteString(routeString);
+
+        return this;
+    }
+
+    /**
+     * Reset class properties
+     *
+     * @for RouteModel
+     * @method reset
+     * @chainable
+     */
+    reset() {
+        this._legCollection = [];
+
+        return this;
+    }
+
+    // ------------------------------ PUBLIC ------------------------------
+
+    // FIXME: COMPLETE THIS METHOD
+    /**
+     * Merge the provided route model into this route model, if possible
+     *
+     * @for RouteModel
+     * @method absorbRouteModel
+     * @param routeModel {RouteModel}
+     */
+    absorbRouteModel(/* routeModel */) {
+        //
+    }
+
+    /**
+     * Calculate the heading from the first waypoint to the second waypoint
+     *
+     * This is used to determine the heading of newly spawned aircraft
+     *
+     * @for RouteModel
+     * @method calculateSpawnHeading
+     * @return {number} heading, in radians
+     */
+    calculateSpawnHeading() {
+        const firstWaypointPositionModel = this.waypoints[0].positionModel;
+        const secondWaypointPositionModel = this.waypoints[1].positionModel;
+        const heading = firstWaypointPositionModel.bearingToPosition(secondWaypointPositionModel);
+
+        return heading;
+    }
+
+    /**
+    * Return an array of waypoints in the flight plan that have altitude restrictions
+    *
+    * @for RouteModel
+    * @method getAltitudeRestrictedWaypoints
+    * @return {array<WaypointModel>}
+    */
+    getAltitudeRestrictedWaypoints() {
```

Comment:
use `filter` in the name here because we are not _getting_ we're filtering 😉

Reply:
no, it should be `filter` 😉 

first, because we are actually using `filter`.  second, because in JS filter means you get something back.

`.filterAltitudeRestrictedWaypoints()` reads exactly as it's doing, which is what we're going for.